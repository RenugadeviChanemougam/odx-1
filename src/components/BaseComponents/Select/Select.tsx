import React from 'react';
import PropTypes from 'prop-types';
import FormGroup from '../FormGroup/FormGroup';

export default function Select(props){

  const {name, onChange, value, children} = props;

  return(
    <FormGroup {...props}>
        <select className="govuk-select" id={name} name={name} onChange={onChange} value={value}>
          {children}
        </select>
    </FormGroup>
  )
}

Select.propTypes = {
  ...FormGroup.propTypes,
  name: PropTypes.string,
  children: PropTypes.node,
  onChange: PropTypes.func,
  value: PropTypes.string,
}
