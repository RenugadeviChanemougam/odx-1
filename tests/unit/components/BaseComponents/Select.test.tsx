import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../../../../src/components/BaseComponents/Select/Select';

it('renders correctly', () => {
  const tree = renderer
    .create(<Select></Select>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const name = "name";
const label = "label";
const hintText = "hintText";
const errorText = "errorText";
const options = [{key:"Option_1", value:"Option 1"}, {key:"Option_2", value:"Option 2"}]
const onChange = () => {};

it('Basic single question select input', () => {
  let a = Select.propTypes;

  const tree = renderer
    .create(<Select name={name} label={label} labelIsHeading={true} onChange={onChange}>
      {options.map(option => {return (<option key={option.key} value={option.value}>{option.value}</option>)})}
    </Select>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Basic multi question select input', () => {
  const tree = renderer
    .create(<Select name={name} label={label} labelIsHeading={false} onChange={onChange}>
      {options.map(option => {return (<option key={option.key} value={option.value}>{option.value}</option>)})}
    </Select>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Basic select input with hint and error text', () => {
    const tree = renderer
    .create(<Select name={name} label={label} onChange={onChange} hintText={hintText} errorText={errorText}>
      {options.map(option => {return (<option key={option.key} value={option.value}>{option.value}</option>)})}
    </Select>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Basic select with pre-selected value', () => {
  const tree = renderer
  .create(<Select name={name} label={label} onChange={onChange} hintText={hintText} value={"Option 2"}>
    {options.map(option => {return (<option key={option.key} value={option.value}>{option.value}</option>)})}
  </Select>)
  .toJSON();
expect(tree).toMatchSnapshot();
});
