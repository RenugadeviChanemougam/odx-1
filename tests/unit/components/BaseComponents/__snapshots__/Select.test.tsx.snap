// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic multi question select input 1`] = `
<div
  className="govuk-form-group"
>
  <label
    className="govuk-label"
    htmlFor="name"
  >
    label
  </label>
  <select
    aria-describedby=""
    className="govuk-select"
    id="name"
    name="name"
    onChange={[Function]}
  >
    <option
      value="Option 1"
    >
      Option 1
    </option>
    <option
      value="Option 2"
    >
      Option 2
    </option>
  </select>
</div>
`;

exports[`Basic select input with hint and error text 1`] = `
<div
  className="govuk-form-group govuk-form-group--error"
>
  <h1
    className="govuk-label-wrapper"
  >
    <label
      className="govuk-label govuk-label--l"
      htmlFor="name"
    >
      label
    </label>
  </h1>
  <div
    className="govuk-hint"
    id="name-hint"
  >
    hintText
  </div>
  <p
    className="govuk-error-message"
    id="name-error"
  >
    <span
      className="govuk-visually-hidden"
    >
      Error:
    </span>
    errorText
  </p>
  <select
    aria-describedby="name-error name-hint"
    className="govuk-select"
    id="name"
    name="name"
    onChange={[Function]}
  >
    <option
      value="Option 1"
    >
      Option 1
    </option>
    <option
      value="Option 2"
    >
      Option 2
    </option>
  </select>
</div>
`;

exports[`Basic select with pre-selected value 1`] = `
<div
  className="govuk-form-group"
>
  <h1
    className="govuk-label-wrapper"
  >
    <label
      className="govuk-label govuk-label--l"
      htmlFor="name"
    >
      label
    </label>
  </h1>
  <div
    className="govuk-hint"
    id="name-hint"
  >
    hintText
  </div>
  <select
    aria-describedby="name-hint"
    className="govuk-select"
    id="name"
    name="name"
    onChange={[Function]}
    value="Option 2"
  >
    <option
      value="Option 1"
    >
      Option 1
    </option>
    <option
      value="Option 2"
    >
      Option 2
    </option>
  </select>
</div>
`;

exports[`Basic single question select input 1`] = `
<div
  className="govuk-form-group"
>
  <h1
    className="govuk-label-wrapper"
  >
    <label
      className="govuk-label govuk-label--l"
      htmlFor="name"
    >
      label
    </label>
  </h1>
  <select
    aria-describedby=""
    className="govuk-select"
    id="name"
    name="name"
    onChange={[Function]}
  >
    <option
      value="Option 1"
    >
      Option 1
    </option>
    <option
      value="Option 2"
    >
      Option 2
    </option>
  </select>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="govuk-form-group"
>
  <h1
    className="govuk-label-wrapper"
  >
    <label
      className="govuk-label govuk-label--l"
    />
  </h1>
  <select
    aria-describedby=""
    className="govuk-select"
  />
</div>
`;
